File: bakery-inventory-backend/src/main/java/com/bakery/inventory/service/StockTransactionService.java
package com.bakery.inventory.service;

import com.bakery.inventory.dto.stocktransaction.StockTransactionResponse;
import java.util.List;

public interface StockTransactionService {
    List<StockTransactionResponse> getAllTransactions();

    List<StockTransactionResponse> getTransactionsByProductId(Integer productId);

    List<StockTransactionResponse> getTransactionsByCategoryId(Integer categoryId);

    List<StockTransactionResponse> getTransactionsByOrderId(Integer orderId);
}

File: bakery-inventory-frontend/src/services/stockTransactionService.js
import axiosInstance from '../api/axiosInstance';

/**
 * NEW FILE: StockTransactionService
 * Fetch audit logs of all inventory transactions.
 */

export const stockTransactionService = {
  getAllTransactions: async () => {
    const response = await axiosInstance.get('/api/stock-transactions');
    return response.data;
  },

  getTransactionsByProductId: async (productId) => {
    const response = await axiosInstance.get(`/api/stock-transactions/product/${productId}`);
    return response.data;
  },

  getTransactionsByCategoryId: async (categoryId) => {
    const response = await axiosInstance.get(`/api/stock-transactions/category/${categoryId}`);
    return response.data;
  },

  getTransactionsByOrderId: async (orderId) => {
    const response = await axiosInstance.get(`/api/stock-transactions/order/${orderId}`);
    return response.data;
  }
};
